### PR TITLE
Bug 1966759: OpenShift plugin adds OCP-specific configuration to projects

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
- internal/plugins/openshift: add OpenShift plugin to add OCP-specificconfiguration to projects
- internal/plugins/{go,ansible,helm}: run openshift/v1 phase 2 init plugin

**Motivation for the change:** some OCP-specific images need to be substituted for upstream ones. A phase 2 plugin accomplishes this for all plugins regardless of origin.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by estroz** [comment]
@jmrodri this is the phase 2 plugin I was talking about.

/cc @jmrodri 

/hold

File: internal/plugins/openshift/v1/plugin.go
Comment (estroz) [inline internal/plugins/openshift/v1/plugin.go:31]: This becomes `openshift.sdk.operatorframework.io`, which isn't really correct. Perhaps something like `openshift.sdk.openshift.io`, which is repetitive but mirrors the structure of the upstream fq name?

File: internal/plugins/openshift/v1/plugin.go
Comment (estroz) [inline internal/plugins/openshift/v1/plugin.go:82]: This may be redundant, since testdata will reflect whether the replacement occurred.

File: patches/04-plugin-openshift-v1
Comment (estroz) [inline patches/04-plugin-openshift-v1:49]: Contents of this file from this point to its end are only necessary if testdata is not regenerated on merge. @jmrodri you said this is the case correct?

File: internal/plugins/openshift/v1/plugin.go
Comment (jmrodri) [reply internal/plugins/openshift/v1/plugin.go:31]: I'm actually okay with `openshift.sdk.operatorframework.io`

File: patches/04-plugin-openshift-v1
Comment (jmrodri) [reply patches/04-plugin-openshift-v1:49]: Yes, we do NOT regenerate the testdata downstream.

**Comment by jmrodri** [comment]
@estroz sanity test failed:

```
tools/scripts/fetch golangci-lint 1.31.0 && tools/bin/golangci-lint run
golangci-lint missing or not version '1.31.0', downloading...
golangci/golangci-lint info checking GitHub for tag 'v1.31.0'
golangci/golangci-lint info found version: 1.31.0 for v1.31.0/linux/amd64
golangci/golangci-lint info installed /go/src/github.com/openshift/ocp-release-operator-sdk/tools/bin/golangci-lint
internal/plugins/openshift/v1/plugin.go:98:45: unnecessary conversion (unconvert)
		if err = ioutil.WriteFile(filePath, []byte(b), info.Mode()); err != nil {
		                                          ^
make: *** [Makefile:123: test-sanity] Error 1
```
You should be able to do the following:
```
make -f ci/prow.Makefile patch
make test-sanity
```

File: internal/plugins/openshift/v1/plugin.go
Comment (camilamacedo86) [inline internal/plugins/openshift/v1/plugin.go:84]: I do not think that will work for the CI. @jmrodri Have we the images when we do the PR?
For the CI, we might need to replace it again.

File: internal/plugins/openshift/v1/plugin.go
Comment (estroz) [reply internal/plugins/openshift/v1/plugin.go]: Thoughts on `sdk.x-openshift.io`? I don't think it's ok to use `operatorframework.io` here since this is technically not an OF plugin.

File: internal/plugins/openshift/v1/init.go
Comment (jmrodri) [inline internal/plugins/openshift/v1/init.go:44]:
```suggestion
// Scaffold updates a newly initialized project with OpenShift-specific configuration.
```

File: internal/plugins/openshift/v1/init.go
Comment (jmrodri) [inline internal/plugins/openshift/v1/init.go:64]: nit: Do we usually do variable initialization this far down? I thought this was a function when I first read this code.  I guess it is closer to the replacement method below so not worth moving.

File: internal/plugins/openshift/v1/init.go
Comment (jmrodri) [inline internal/plugins/openshift/v1/init.go:100]: stating is the present participle of state. So that reads pretty weird, why are you "state - ing" a file? :)

```suggestion
			return fmt.Errorf("error trying to stat file for substitution: %v", err)
```

**Review by jmrodri — COMMENTED** [review]
Just a couple items, comment seems wrong and "stating" error reads weird. The nit of variable initialization is just a nit, leave it alone.

**Comment by jmrodri** [comment]
We need to hold this until PR #114 merges, which will cause this PR to rebase (probably).

File: internal/plugins/openshift/v1/init.go
Comment (estroz) [reply internal/plugins/openshift/v1/init.go:64]: It's a matter of preference, but my preference is to define globals used by few/one function/method just above or below those functions.

**Comment by emmajiafan** [comment]
ansible type operator:

**Review by jmrodri — APPROVED** [review]
/lgtm

**Comment by jmrodri** [comment]
/retitle Bug 1966759: OpenShift plugin adds OCP-specific configuration to projects

**Pull request merged**
